streams/src/test/java/org/apache/kafka/test/KStreamTestDriver.java

```diff
@@ -109,10 +109,7 @@ public ProcessorContext context() {
 
     public void process(String topicName, Object key, Object value) {
         final ProcessorNode prevNode = context.currentNode();
-        ProcessorNode currNode = topology.source(topicName);
-        if (currNode == null && globalTopology != null) {
-            currNode = globalTopology.source(topicName);
-        }
+        ProcessorNode currNode = nodeByTopicName(topicName);
```

Comment:
Nit: add `final`
Can you also update the three method parameters and add `final` to each please?